Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -1008,6 +1008,81 @@ public void Restart()
 
         protected abstract void RestartInternal();
 
+        /// <summary>
+        /// Comparison function used in <see cref="GetLatestDLLVersion" />.
+        /// </summary>
+        private static int VersionCompare(Tuple<Version, string, string> a, Tuple<Version, string, string> b)
```

Comment:
This should work just fine: https://docs.microsoft.com/en-us/dotnet/api/system.version.compareto?view=netcore-3.1

Reply:
The three values are for sorting purposes.

1st is the version
2nd is the shortened folder name (minus the version)
3rd is the complete folder.

the sorting is by short name then by version.

That way old unversioned folder get matched as well.